### PR TITLE
new camera profile to support full res aux camara, when available

### PR DESCRIPTION
**Pull request by psmithcrl**
Camera Profile for full res color camera support.

### CODE REVIEW

File: source/LibMultiSense/MultiSenseTypes.hh [hunk 1]
```diff
@@ -168,6 +168,8 @@ static CRL_CONSTEXPR CameraProfile High_Contrast = (1U<<1);
 static CRL_CONSTEXPR CameraProfile Show_ROIs = (1U<<2);
 /** User would like to run spline-based ground surface algorithm on the camera*/
 static CRL_CONSTEXPR CameraProfile Ground_Surface = (1U<<3);
+/** User would like full resolution color camera while reducing the stereo pair to 1/4 resolution*/
```
Comment (mattalvarado) [inline source/LibMultiSense/MultiSenseTypes.hh:171]: I would update this to something like:
`User would like full resolution images from the aux camera regardless of the requested resolution of the stereo pair`

**Review by pdlr — COMMENTED** [review]
Is a profile the best way to do this?  I imagine that expicitly setting the resolution would have the most downstream flexibility, but perhaps there's a technical reason that the profile mechanism works better?

File: source/LibMultiSense/MultiSenseTypes.hh (hunk 1 above)
Comment (psmithcrl) [reply source/LibMultiSense/MultiSenseTypes.hh:171]: I like that wording. Thanks. I'll change it now.

**Comment by psmithcrl** [comment]
> Is a profile the best way to do this? I imagine that expicitly setting the resolution would have the most downstream flexibility, but perhaps there's a technical reason that the profile mechanism works better?
I agree a profile may not be the best way to implement this, but we currently don't have a mechanism to adjust the resolution of the aux camera, independent of the stereo pair.

**Review by pdlr — APPROVED** [review]

File: source/LibMultiSense/MultiSenseTypes.hh [hunk 2]
```diff
@@ -1005,6 +1007,11 @@ public:
      */
     void setCameraProfile(const CameraProfile &profile)
     {
+        if (profile == Full_Res_Aux_Cam)
+        {
+            fprintf(stderr, "Warning: Deprecated Camera Profile: Full_Res_Aux_Cam.\n");
+            fprintf(stderr, "         Will be changed in future revisions.\n");
+        }
```
Comment (mattalvarado) [inline source/LibMultiSense/MultiSenseTypes.hh:1014]: What about this will be depreciated?

**Review by mattalvarado — APPROVED** [review]

File: source/LibMultiSense/MultiSenseTypes.hh (hunk 2 above)
Comment (psmithcrl) [reply source/LibMultiSense/MultiSenseTypes.hh]: In the future if we decide to change the interface to this function, then the profile itself will be deprecated.
However, I don't think the feature itself will be deprecated.
I have removed the print statement in favor of placing the warning in the documentation.

**Pull request merged**